packages/task/src/browser/task-configurations.ts

```diff
@@ -401,7 +399,11 @@ export class TaskConfigurations implements Disposable {
                 }
                 const transformedTask = this.getTransformedRawTask(taskConfig, scope);
                 if (this.isDetectedTask(transformedTask)) {
-                    addCustomization(scopeKey, transformedTask);
+                    // TODO: `getTransformedRawTask` is it safe to assume `transformedTask` is a `TaskConfiguration`?
+                    //       - all branches in `getTransformedRawTask` seem to add the `_scope` property.
+                    //       - the `else` branch of this if statement seems to assume a `label` property exists.
+                    //       - These two properties are what `TaskConfiguration` adds "on top" of `TaskCustomization`...
+                    addCustomization(scopeKey, transformedTask as TaskConfiguration);
```

Comment:
If it seems like what is returned is always going to be a `TaskConfiguration`, then it would be safer to change the return type of `getTransformedRawTask` rather than to recast the return here. That way TS will help us ensure that this assumption is and remains correct.

Reply:
Hello @colin-grant-work 

I'm not 100% sure it is safe to make this assumption.
Which is why I left this `TODO` for the review discussion...

I can see that the `_scope` property will always be added inside `getTransformedRawTask()`
But I am not sure about the existence of the `label` property.

Reply:
One case in which label wouldn't be defined would be the following:  
1. Contribute a task type using Theia's internal mechanisms
2. Define a customization of that task type in a `tasks.json` without  a `label` field - maybe having no label is OK for that task type if it isn't supposed to be accessible by the task enumerating quick-picks.

It seems like that's the scenario that this code is guarding against. The assumption seems to be something like this: labels are for 'instance' task types (i.e. `TaskConfiguration`), where the user is in control of all the important paramaters, but in cases where the user only tweaks a few parameters (i.e. `TaskCustomization`), the `type` field itself is the identifier, and there _shouldn't_ be multiple instances that need their own identifiers. Those assumptions may not be correct - I'm not very familiar with the contributed task system, and this is likely something that @tsmaeder could speak to in greater detail - but I think that's what's behind the distinction here.